consumer/src/main/kotlin/streams/kafka/KafkaEventSink.kt

```diff
@@ -113,15 +148,172 @@ class KafkaEventConsumer(private val consumer: KafkaConsumer<String, ByteArray>,
+
+    override fun start() {
+        if (topics.isEmpty()) {
+            log.info("No topics specified Kafka Consumer will not started")
+            return
+        }
+        this.consumer.subscribe(topics, StreamsConsumerRebalanceListener(topicPartitionOffsetMap, consumer, config.autoOffsetReset, log))
+    }
+
+    private fun readSimple(action: (String, List<Any>) -> Unit): Map<TopicPartition, OffsetAndMetadata> {
+        val topicMap = mutableMapOf<TopicPartition, OffsetAndMetadata>()
+        val records = consumer.poll(0)
+        if (!records.isEmpty) {
+            this.topics.forEach { topic ->
+                val topicRecords = records.records(topic)
+                val lastRecord = topicRecords.last()
+                val offsetAndMetadata = OffsetAndMetadata(lastRecord.offset(), "")
+                val topicPartition = TopicPartition(lastRecord.topic(), lastRecord.partition())
+                topicMap[topicPartition] = offsetAndMetadata
+                topicPartitionOffsetMap[topicPartition] = offsetAndMetadata
+                try {
+                    action(topic, topicRecords.map { JSONUtils.readValue<Any>(it.value()) })
+                } catch (e: Exception) {
+                    // TODO add dead letter queue
+                }
+            }
+        }
+        return topicMap
+    }
+
+    private fun readFromPartition(kafkaTopicConfig: KafkaTopicConfig,
+                                  action: (String, List<Any>) -> Unit): Map<TopicPartition, OffsetAndMetadata> {
+        val topicMap = mutableMapOf<TopicPartition, OffsetAndMetadata>()
+        setSeek(kafkaTopicConfig.topicPartitionsMap)
+        val records = consumer.poll(0)
+        val consumerRecordsMap = toConsumerRecordsMap(kafkaTopicConfig.topicPartitionsMap, records)
+        if (consumerRecordsMap.isNotEmpty()) {
+            try {
+                consumerRecordsMap.forEach {
+                    val lastRecord = it.value.last()
+                    val topicPartition = TopicPartition(lastRecord.topic(), lastRecord.partition())
+                    val offsetAndMetadata = OffsetAndMetadata(lastRecord.offset(), "")
+                    topicMap[topicPartition] = offsetAndMetadata
+                    topicPartitionOffsetMap[topicPartition] = offsetAndMetadata
+                    action(it.key.topic(), it.value.map { JSONUtils.readValue<Any>(it.value()) })
+                }
+            } catch (e: Exception) {
+                // TODO add dead letter queue
+            }
+        }
+        return topicMap
+    }
+
+    private fun commitData(commit: Boolean, topicMap: Map<TopicPartition, OffsetAndMetadata>) {
+        if (commit && topicMap.isNotEmpty()) {
+            consumer.commitSync(topicMap)
+        }
+    }
+
+    override fun read(topicConfig: Map<String, Any>, action: (String, List<Any>) -> Unit) {
+        val kafkaTopicConfig = KafkaTopicConfig.fromMap(topicConfig)
```

Comment:
do we have to do this for every read? wouldn't it make more sense to only compute it once? I guess the config doesn't change so much?

Only if we want to make the config reloadable etc.